File: app/src/main/java/hr/ml/moviesdiscover/rest/model/MovieDetailsFromRetrofit.java
package hr.ml.moviesdiscover.rest.model;

import java.util.List;

public class MovieDetailsFromRetrofit {
    private boolean adult;
    private String backdropPath;
    private Object belongsToCollection;
    private int budget;
    private List<GenreFromRetrofit> genres = null;
    private String homepage;
    private int id;
    private String imdbId;
    private String originalLanguage;
    private String originalTitle;
    private String overview;
    private float popularity;
    private String posterPath;
    private List<ProductionCompanyFromRetrofit> productionCompanies = null;
    private List<ProductionCompanyFromRetrofit> productionCountries = null;
    private String releaseDate;
    private int revenue;
    private int runtime;
    private List<SpokenLanguageFromRetrofit> spokenLanguages = null;
    private String status;
    private String tagline;
    private String title;
    private boolean video;
    private float voteAverage;
    private int voteCount;

    public boolean isAdult() {
        return adult;
    }

    public void setAdult(boolean adult) {
        this.adult = adult;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public void setBackdropPath(String backdropPath) {
        this.backdropPath = backdropPath;
    }

    public Object getBelongsToCollection() {
        return belongsToCollection;
    }

    public void setBelongsToCollection(Object belongsToCollection) {
        this.belongsToCollection = belongsToCollection;
    }

    public int getBudget() {
        return budget;
    }

    public void setBudget(int budget) {
        this.budget = budget;
    }

    public List<GenreFromRetrofit> getGenres() {
        return genres;
    }

    public void setGenres(List<GenreFromRetrofit> genres) {
        this.genres = genres;
    }

    public String getHomepage() {
        return homepage;
    }

    public void setHomepage(String homepage) {
        this.homepage = homepage;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getImdbId() {
        return imdbId;
    }

    public void setImdbId(String imdbId) {
        this.imdbId = imdbId;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public void setOriginalLanguage(String originalLanguage) {
        this.originalLanguage = originalLanguage;
    }

    public String getOriginalTitle() {
        return originalTitle;
    }

    public void setOriginalTitle(String originalTitle) {
        this.originalTitle = originalTitle;
    }

    public String getOverview() {
        return overview;
    }

    public void setOverview(String overview) {
        this.overview = overview;
    }

    public float getPopularity() {
        return popularity;
    }

    public void setPopularity(float popularity) {
        this.popularity = popularity;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public void setPosterPath(String posterPath) {
        this.posterPath = posterPath;
    }

    public List<ProductionCompanyFromRetrofit> getProductionCompanies() {
        return productionCompanies;
    }

    public void setProductionCompanies(List<ProductionCompanyFromRetrofit> productionCompanies) {
        this.productionCompanies = productionCompanies;
    }

    public List<ProductionCompanyFromRetrofit> getProductionCountries() {
        return productionCountries;
    }

    public void setProductionCountries(List<ProductionCompanyFromRetrofit> productionCountries) {
        this.productionCountries = productionCountries;
    }

    public String getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(String releaseDate) {
        this.releaseDate = releaseDate;
    }

    public int getRevenue() {
        return revenue;
    }

    public void setRevenue(int revenue) {
        this.revenue = revenue;
    }

    public int getRuntime() {
        return runtime;
    }

    public void setRuntime(int runtime) {
        this.runtime = runtime;
    }

    public List<SpokenLanguageFromRetrofit> getSpokenLanguages() {
        return spokenLanguages;
    }

    public void setSpokenLanguages(List<SpokenLanguageFromRetrofit> spokenLanguages) {
        this.spokenLanguages = spokenLanguages;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public String getTagline() {
        return tagline;
    }

    public void setTagline(String tagline) {
        this.tagline = tagline;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public boolean isVideo() {
        return video;
    }

    public void setVideo(boolean video) {
        this.video = video;
    }

    public float getVoteAverage() {
        return voteAverage;
    }

    public void setVoteAverage(float voteAverage) {
        this.voteAverage = voteAverage;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public void setVoteCount(int voteCount) {
        this.voteCount = voteCount;
    }
}

File: app/src/main/kotlin/hr/ml/moviesdiscover/rest/model/MovieDetailsFromRetrofit.kt
package hr.ml.moviesdiscover.rest.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class MovieDetailsFromRetrofit(

        @SerializedName("adult")
        @Expose
        val adult: Boolean,

        @SerializedName("backdrop_path")
        @Expose
        val backdropPath: String?,

        @SerializedName("belongs_to_collection")
        @Expose
        val belongsToCollection: Any?,

        @SerializedName("budget")
        @Expose
        val budget: Int,

        @SerializedName("genres")
        @Expose
        val genres: List<GenreFromRetrofit>?,

        @SerializedName("homepage")
        @Expose
        val homepage: String?,

        @SerializedName("id")
        @Expose
        val id: Int,

        @SerializedName("imdb_id")
        @Expose
        val imdbId: String?,

        @SerializedName("original_language")
        @Expose
        val originalLanguage: String?,

        @SerializedName("original_title")
        @Expose
        val originalTitle: String?,

        @SerializedName("overview")
        @Expose
        val overview: String?,

        @SerializedName("popularity")
        @Expose
        val popularity: Float,

        @SerializedName("poster_path")
        @Expose
        val posterPath: String?,

        @SerializedName("production_companies")
        @Expose
        val productionCompanies: List<ProductionCompanyFromRetrofit>?,

        @SerializedName("production_countries")
        @Expose
        val productionCountries: List<ProductionCompanyFromRetrofit>?,

        @SerializedName("release_date")
        @Expose
        val releaseDate: String?,

        @SerializedName("revenue")
        @Expose
        val revenue: Int,

        @SerializedName("runtime")
        @Expose
        val runtime: Int,

        @SerializedName("spoken_languages")
        @Expose
        val spokenLanguages: List<SpokenLanguageFromRetrofit>?,

        @SerializedName("status")
        @Expose
        val status: String?,

        @SerializedName("tagline")
        @Expose
        val tagline: String?,

        @SerializedName("title")
        @Expose
        val title: String?,

        @SerializedName("video")
        @Expose
        val video: Boolean,

        @SerializedName("vote_average")
        @Expose
        val voteAverage: Float,

        @SerializedName("vote_count")
        @Expose
        val voteCount: Int,

)